Tests/ArrowCsvForeignStorageTest.cpp

```diff
@@ -297,6 +387,26 @@ TEST(DecimalDataTest, DifferentSizesOfDecimal) {
                        {999.9, 9384612.78, 2947583746581.92748}});
 }
 
+TEST(DecimalTest, FragmentedDecimal) {
+  run_ddl_statement("CREATE DATAFRAME decimal_dataframe" + decimal_table_format +
+                    " FROM "
+                    "'CSV:../../Tests/Import/datafiles/santander_top1000.csv'");
+  run_ddl_statement("TRUNCATE TABLE decimal_table");
+  run_ddl_statement(
+      "COPY decimal_table FROM '../../Tests/Import/datafiles/santander_top1000.csv'");
+
+  std::string col_names = "";
+  const size_t n_decimal_cols = 200;
+
+  for (size_t i = 0; i < n_decimal_cols - 1; ++i) {
+    col_names += "var_" + std::to_string(i) + ", ";
+  }
+  col_names += "var_" + std::to_string(n_decimal_cols - 1);
+
+  check_tables<double>("SELECT " + col_names + " FROM decimal_dataframe",
```

Comment:
please also add tests for groupby, where closure and other.
add fragment_size here, even if it would fail because of bug.